// presentation/presentation.js
/**
 * Created by Phil on 2015-02-20.
 */

var indice = '1';
var varPresentationArray = [];
var varAddrIndTimeout = [];

module.exports = {
    getVar: function(indice){
        //console.log(indice);
        for(var i = 0; i < varAddrIndTimeout.length; i++){
            if(indice == varAddrIndTimeout[i][1]){
                varAddrIndTimeout[i][2] = 6;
                return varPresentationArray[i];
            }
        }
        return -1;

    },

    resetTimeout: function(indice){
        for(var i = 0; i < varAddrIndTimeout.length; i++){
            if(indice == varAddrIndTimeout[i][1]){
                varAddrIndTimeout[i][2] = 6;
            }
        }
    },

    setVar: function(varPre, indice, ipAddr){
        //console.log(indice);
        //console.log(varPre);
        var ind = 0;
        for(var i = 0; i < varAddrIndTimeout.length; i++){
            if(indice == varAddrIndTimeout[i][1]){
                //console.log('trouve');
                ind = i;
            }
        }

        if(ipAddr == varAddrIndTimeout[ind][0]) {
            if ((ind >= 0) && (ind < varPresentationArray.length)) {
                varPresentationArray[ind].varUtilisateur = varPre.varUtilisateur;
                varPresentationArray[ind].varNomAnimateur = varPre.varNomAnimateur;
                varPresentationArray[ind].varNomStation = varPre.varNomStation;
                varPresentationArray[ind].varCapacite = varPre.varCapacite;
                varPresentationArray[ind].varVilles = varPre.varVilles;
                varPresentationArray[ind].varPopulation = varPre.varPopulation;
                varPresentationArray[ind].varProvenance = varPre.varProvenance;
                varPresentationArray[ind].varNomPlanDEau = varPre.varNomPlanDEau;
                varPresentationArray[ind].varParticularite = varPre.varParticularite;
                varPresentationArray[ind].varProcedes = varPre.varProcedes;
                varPresentationArray[ind].varMapsLink = varPre.varMapsLink;
                varPresentationArray[ind].varOrdrePresentation = varPre.varOrdrePresentation;
                varPresentationArray[ind].varNomImage = varPre.varNomImage;
                varPresentationArray[ind].varDateCreation = varPre.varDateCreation;
                varPresentationArray[ind].varDateModif = varPre.varDateModif;
                varPresentationArray[ind].varDateAcces = varPre.varDateAcces;
                varPresentationArray[ind].varEtat = varPre.varEtat;
                varAddrIndTimeout[ind][2] = 6;
                return 0;
            } else {
                return -1;
            }
        }else{
            return -2;
        }

    },

    addVar: function(ipAddr){
        var newIndice = indice;
        var addrArr = [ipAddr,newIndice,6];
        var varPresentation = {
            varUtilisateur: '',
            varNomAnimateur: '',
            varNomStation: '',
            varCapacite: '',
            varVilles: '',
            varPopulation: '',
            varProvenance: '',
            varNomPlanDEau: '',
            varParticularite: [],
            varProcedes: [],
            varMapsLink: [],
            varOrdrePresentation: [],
            varNomImage: '',
            varDateCreation: '',
            varDateModif: '',
            varDateAcces: '',
            varEtat: ''
        };
        varAddrIndTimeout.push(addrArr);
        varPresentationArray.push(varPresentation);
        //console.log(varPresentationArray[0]);
        indice = (parseInt(indice)+1).toString();
        return newIndice;
    },

    resetVar: function(indice, ipAddr){
        var ind = 0;
        for(var i = 0; i < varAddrIndTimeout.length; i++){
            if(indice == varAddrIndTimeout[i][1]){
                ind = i;
            }
        }
        if(ipAddr == varAddrIndTimeout[ind][0]) {
            if ((ind >= 0) && (ind < varPresentationArray.length)) {
                //varPresentationArray[ind].varUtilisateur = "";
                varPresentationArray[ind].varNomAnimateur = "";
                varPresentationArray[ind].varNomStation = "";
                varPresentationArray[ind].varCapacite = "";
                varPresentationArray[ind].varVilles = "";
                varPresentationArray[ind].varPopulation = "";
                varPresentationArray[ind].varProvenance = "";
                varPresentationArray[ind].varNomPlanDEau = "";
                varPresentationArray[ind].varParticularite = [];
                varPresentationArray[ind].varProcedes = [];
                varPresentationArray[ind].varMapsLink = [];
                varPresentationArray[ind].varOrdrePresentation = [];
                varPresentationArray[ind].varNomImage = "";
                varPresentationArray[ind].varDateCreation = "";
                varPresentationArray[ind].varDateModif = "";
                varPresentationArray[ind].varDateAcces = "";
                varPresentationArray[ind].varEtat = "";
                varAddrIndTimeout[ind][2] = 6;
                return 0;
            } else {
                return -1;
            }
        }else{
            return -2;
        }
    },

    verifyVarTimeout: function(){
        var varTimeout = [];
        //console.log(varAddrIndTimeout);
        //console.log(varPresentationArray);
        if(varAddrIndTimeout.length > 0) {
            for (var i = 0; i < varAddrIndTimeout.length; i++) {
                varAddrIndTimeout[i][2] = varAddrIndTimeout[i][2] - 1;
                //console.log(varAddrIndTimeout[i][2]);
                if (varAddrIndTimeout[i][2] <= 0) {
                    //console.log('condition');
                    varTimeout.push(varAddrIndTimeout[i][1]);
                }
            }
            //console.log(varTimeout);
            for(var j = 0; j < varTimeout.length; j++){
                for(var k = 0; k < varAddrIndTimeout.length; k++){
                    if(varAddrIndTimeout[k][1] == varTimeout[j]) {
                        varAddrIndTimeout.splice(k, 1);
                        varPresentationArray.splice(k, 1);
                    }
                }
            }
        }else{
            indice = '1';
        }
    }
}
